apps/dg/components/graph/plots/case_plot_view.js

```diff
@@ -160,53 +160,152 @@ DG.CasePlotView = DG.PlotView.extend(
+                        tIsDragging = true;
+                        this.ox = this.attr("cx");
+                        this.oy = this.attr("cy");
+                        this.animate({opacity: kOpaque}, DG.PlotUtilities.kDataTipShowTime, "bounce");
+                        this.toFront();
+                      },
+                      function () {  // end
+                        this.animate({transform: tInitialTransform}, DG.PlotUtilities.kHighlightHideTime);
+                        tIsDragging = false;
+                      })
+              ;
+            }.bind(this),
 
-        var tCases = this.getPath('model.cases'),
-            tPlottedElements = this.get('plottedElements'),
-            tRC = this.createRenderContext();
+            tXPixelMin = this.getPath('xAxisView.pixelMin'),
+            tXPixelMax = this.getPath('xAxisView.pixelMax'),
+            tYPixelMax = this.getPath('yAxisView.pixelMin'),
+            tYPixelMin = this.getPath('yAxisView.pixelMax'),
 
-        this._pointRadius = this.calcPointRadius(); // make sure created circles are of right size
-        this.setPath('model.isAnimating', true); // So the animation can finish
-        animatePoints();  // will loop through all points using invokeLater
+            tPointSet,
+            tPlottedElements = this.get('plottedElements'),
+            tModel = this.get('model'),
+            tNumCases = tModel.get('cases').length(),
+            kMargin = 10,
+            tR = this.calcPointRadius(),
+            tColor = tModel.getPointColor ? tModel.getPointColor() : DG.PlotUtilities.kDefaultPointColor,
+            tStrokeParams = this.getStrokeParams();
+        if (!tPlottedElements || tPlottedElements.length === 0) {
+          var tStart = 0,
+              tInc = 200,
+              loop = function () {
+                var tDescriptions = [];
+                tXPixelMax = this.getPath('xAxisView.pixelMax');
+                tYPixelMax = this.getPath('yAxisView.pixelMin');
+                if( tXPixelMax < 30 || tYPixelMax < 30) {
+                  this.invokeLater( loop, 100);
+                  return;
+                }
+                for (var i = tStart; i < tStart + tInc && i < tNumCases; i++) {
+                  tDescriptions.push(createCircleDescription(i));
+                }
+                tPointSet = this.get('paper').add(tDescriptions).attr({
+                  r: tR / 2,
+                  fill: 'yellow',
+                  stroke: tStrokeParams.strokeColor,
+                  'stroke-opacity': tStrokeParams.strokeTransparency,
+                  cursor: 'pointer'
+                }).forEach(function (iElement, iIndex) {
+                  tPlottedElements.push(iElement);
+                  iElement.addClass(DG.PlotUtilities.kDotClassName);
+                  iElement.index = iIndex;
+                  iElement.node.setAttribute('shape-rendering', 'geometric-precision');
+                  return true;
+                });
```

Comment:
I am curious why a number of these things were not set in the json object during createCircleDescription(). Did you try it and find no performance improvement? I ask because it looks to me like Raphael spends a great deal of time in a loop after it gets control and before it starts rendering.

Reply:
Well, I think you are right. I'll give it a try.